iroh-net/src/netcheck/reportgen.rs

```diff
@@ -599,42 +605,44 @@ impl Actor {
                 let netcheck = self.netcheck.clone();
                 let pinger = pinger.clone();
 
-                set.push(Box::pin(async move {
-                    run_probe(
-                        reportstate,
-                        stun_sock4,
-                        stun_sock6,
-                        derp_node,
-                        probe,
-                        netcheck,
-                        pinger,
-                    )
-                    .await
-                }));
+                set.spawn(run_probe(
+                    reportstate,
+                    stun_sock4,
+                    stun_sock6,
+                    derp_node,
+                    probe,
+                    netcheck,
+                    pinger,
+                ));
             }
 
             // Add the probe set to all futures of probe sets.  Handle aborting a probe set
             // if needed, only normal errors means the set continues.
-            probes.push(Box::pin(async move {
+            probes.spawn(async move {
                 // Hack because ProbeSet is not it's own type yet.
                 let mut probe_proto = None;
-                while let Some(res) = set.next().await {
+                while let Some(res) = set.join_next().await {
                     match res {
-                        Ok(report) => return Ok(report),
-                        Err(ProbeError::Error(err, probe)) => {
+                        Ok(Ok(report)) => return Ok(report),
+                        Ok(Err(ProbeError::Error(err, probe))) => {
                             probe_proto = Some(probe.proto());
                             warn!(?probe, "probe failed: {:#}", err);
                             continue;
                         }
-                        Err(ProbeError::AbortSet(err, probe)) => {
+                        Ok(Err(ProbeError::AbortSet(err, probe))) => {
                             debug!(?probe, "probe set aborted: {:#}", err);
+                            set.abort_all();
                             return Err(err);
                         }
+                        Err(err) => {
+                            warn!("fatal probe set error, aborting: {:#}", err);
+                            return Err(anyhow::anyhow!(err));
+                        }
```

Comment:
If a single probe panics (or is cancelled) I think it's probably fine to continue the other probes.  Not really sure if all of them would panic as they run the same code... but really none should panic so I guess I'd still choose to carry on.

Not sure if we can log the panic in a reasonable way though.

Reply:
kept the logging, can `continue` now